packages/stacks-hooks/src/s3-hook.ts

```diff
@@ -0,0 +1,40 @@
+import { Hook, HookInput, HookOutput } from "@takomo/stacks-model"
+import { uuid } from "@takomo/util"
+
+interface S3HookProps {
+  readonly bucket: string
+  readonly key: string
+  readonly content: any
+  readonly iamRoleArn?: string
+}
+
+export class S3Hook implements Hook {
+  readonly bucket: string
+  readonly key: string
+  readonly content: string
+  readonly iamRoleArn?: string
+  constructor({ bucket, key, content, iamRoleArn }: S3HookProps) {
+    this.bucket = bucket
+    this.key = key
+    this.content = JSON.stringify(content)
+    this.iamRoleArn = iamRoleArn
+  }
+
+  async execute({ ctx, logger, stack }: HookInput): Promise<HookOutput> {
+    const crManager = this.iamRoleArn
+      ? await ctx.credentialManager.createCredentialManagerForRole(
+          this.iamRoleArn,
+        )
+      : ctx.credentialManager
+    const s3Client = await ctx.awsClientProvider.createS3Client({
+      logger,
+      credentialProvider: crManager.getCredentialProvider(),
+      id: uuid(),
+      region: stack.region,
```

Comment:
Should we pass the `region` in hook properties. I'm not sure how the put operation works if the s3 client's region is different that the bucket's region. By default, we can use stack's region if no region is given in the hook properties.